api/tests/logic/user_test.py

```diff
@@ -267,3 +269,42 @@ def test_add_preferences_adds_multiple_on_opt_in(session, subscription):
     user = User.query.filter(User.id == user.id).one()
     assert user.subscription_preferences[0].preference in (preference_1, preference_2)
     assert user.subscription_preferences[1].preference in (preference_1, preference_2)
+
+
+def is_valid_user_subscription_preference_no_subscription(subscription):
+    preference = subscription.datetime[0]
+    user = User(email='a@yelp.com', meta_data={'department': 'dept'})
+    user_sub = UserSubscriptionPreferences(preference=preference, subscription_id=None, user=user)
+    subscription.user_rules = [Rule(name='department', value='dept')]
+
+    result = is_valid_user_subscription_preference(user_sub, None)
+    assert not result
+
+
+def is_valid_user_subscription_preference_user_terminated(subscription):
```

Comment:
I might be missing something here. How is this test user set to be terminated?

Reply:
you are not missing something, I forgot to add `terminated=True` to the test.